docs/DeveloperGuide.md

```diff
@@ -1,38 +1,726 @@
+
+#### Rationale
+`WellNus` directly transfers control of user input to `MainManager.runEventDriver()` as managing user input is the
+expected functionality of the `runEventDriver()` method within a particular implementation of `Manager`, which means
+that conceptually, management of user input belongs in a subclass of `Manager` instead. Besides, this abstraction
+of user input logic from `WellNus` fulfils the `Single Responsibility Principle` since `WellNus` is intended
+to be a high-level class that delegates tasks to specialised classes that provide the expected functionality, and thus
+`WellNus` must not be responsible for concrete logic such as managing user input.
+
+Additionally, `MainManager.runEventDriver()` is intentionally restricted to only recognise basic commands and feature
+keywords to firstly, achieve the encapsulation and abstraction of feature-specific logic from `MainManager`. Moving 
+feature-specific logic such as recognising feature-specific commands to corresponding feature `Managers` ensures that
+actual implementation details in feature-related subpackages are hidden from `MainManager`. This is necessary for 
+the purpose of encapsulation since `MainManager` exists in a different subpackage. At the same time, by providing the
+public `runEventDriver()` in feature `Managers`, `MainManager` is only aware of the expected functionality of the
+`runEventDriver()`, which can be used to support feature-specific commands, without being involved in the implementation
+details. This allows `MainManager.runEventDriver()` to be kept abstract while providing the expected functionality of
+the application. Secondly, this design fulfils the `Single Responsibility Principle` as `MainManager` is solely
+responsible for the main WellNUS++ commands but not any feature-specific ones, which means that its logic will only
+be changed for reasons related to the main WellNUS++ commands only.
+
+Lastly, the `runEventDriver()` method of feature `Managers` delegates the execution of commands to implementations of
+`Command` to abide by the `Single Responsibility Principle`. Every `Manager.runEventDriver()` method is expected to
+provide a particular user interface, but not any commands. This means that this method should only change for reasons
+related to its user interface, which requires that command handling logic be implemented elsewhere so that changes in
+commands do not require changes in any implementation of `Manager.runEventDriver()`. Besides, this approach ensures
+abstraction of logic as `Manager.runEventDriver()` ensures that command handling is performed while avoiding the
+actual implementation details by delegating the task to a particular implementation of `Command.execute()`, which is
+known to provide command handling functionality.
+
+### Reflection Component
+
+![Reflection Component Class Diagram](diagrams/ReflectionSequenceDiagram.png)
+![Reflection Component Class Diagram](diagrams/ReflectionClassDiagram.png)
```

Comment:
The class diagram here could possibly be abstracted out and separated out into different features since it is not mandatory to include everything. Users may find it hard to read as the text size is a little small